### ISSUE TITLE
ghc -ddump-simpl-stats output missing

### ISSUE THREAD

**Issue by harendra-kumar** [commercialhaskell/stack#2012]
I used the following command line to print the ghc simplifier stats:

```
stack build --ghc-options -ddump-simpl-stats
```

I also tried the `-v` option but I could not find the output of the command anywhere. Neither on console nor in the logs. Though when I invoked ghc directly I see the output as follows. I expect to find the same with `stack build` as well.

```
cueball:/vol/hosts/cueball/workspace/projects/unicode-transforms/Data/Text$ stack ghc --package unicode-transforms -- -ddump-simpl-stats NormalizeNative.hs
[1 of 1] Compiling Data.Text.NormalizeNative ( NormalizeNative.hs, NormalizeNative.o )

==================== Grand total simplifier statistics ====================
Total ticks:     2

1 PreInlineUnconditionally 1 t_a2Uk
1 BetaReduction 1 t_a2Uk
2 SimplifierDone 2
```

**Comment by mgsloan** [comment]
This is likely due to stack providing the ghc options `-ddump-hi -ddump-to-file` so that it can get build dependency info.  Not sure what to about that, except perhaps putting the file to the console or something (but without the ddump-hi portions)

**Comment by harendra-kumar** [comment]
Even a pointer to the dump file location at the end of the compilation should help. At least the user will know what happened and where the info can be found.

**Comment by mgsloan** [comment]
Yeah, that's a good solution!

**Comment by harendra-kumar** [comment]
Another problem when using `-ddump` options is that stack keeps accumulating the dump to the same files. It becomes very difficult to read the core/stg/cmm when the dump has output collected from multiple runs. We should perhaps overwrite the files every time. @mgsloan is that easily possible?

**Comment by mgsloan** [comment]
It's strange to me that the ddump output appends, that may be an oversight in ghc, because for the `dump-hi` files it recreates them.

It'd be tricky to correctly clear out the dump files.  We can either:

1) Remove all dump files.  This means that for partial recompiles, we'd only have dump files for stuff that recompiled.  So to get all ddump data you'd need to do `-fforce-recomp`.

2) Somehow know which modules will recompile, and delete the associated dump files.

**Comment by harendra-kumar** [comment]
Earlier I thought it appends. But it does not append it just appears twice in each file every time. Strangely it does not happen for .dump-simpl but happens for all other dump files. Perhaps some ghc option is causing it, not sure which one.

**Comment by sjakobi** [comment]
> Even a pointer to the dump file location at the end of the compilation should help.

This idea is also being tracked in https://github.com/commercialhaskell/stack/issues/1727.

**Comment by snoyberg** [comment]
Closing in favor of #1727. Also, this should be addressed more completely by #4545
